### ISSUE TITLE
Embedded presentations, two issues, single fix

### ISSUE THREAD

**Issue by Martinomagnifico** [hakimel/reveal.js#3080]
* Issue 1: **In all browsers**, when we have any embedded presentation, remotely opening it in fullscreen won't let you use the keyboard immediately, only after another click. When the controls are set to false, as in some presentations, the user may not understand why it does not work. See the attached example of using remote triggers.
* Issue 2: **In Safari**, when we have any embedded presentation, selecting it and pressing 'F' will NOT scale the first slide initially, only after navigating. This is even visible in the main top presentation on revealjs.com, and in the "multiple-presentations.html" in the examples. 
* (Issue 3): **In Safari**, in some of the embedded of the examples, like on this page: https://revealjs.com/fullscreen/ and the multiple presentation example, fullscreen turns into a bar, not my idea of fullscreen. This is a combination of issue 2 and the flexbox markup.

Here's the fix:
Storing the `keyboardCondition` and temporarily setting it to null fixes issue 1 and 2. At first, using `layout()` on the deck inside the `fullscreenchange` function fixed issue 2 for me, but it seems that it is no longer needed when using the `keyboardCondition` fix of issue 1. Maybe changing the config automatically triggers a new layout, not sure about that. Issue 3 is fixed with some change to the markup/CSS compared to the original multiple presentations page.

Below is an example of multiple presentations that can be triggered by remote buttons. It includes the fixes above in the right-side presentation.  I added Screenfull to be able to assign the remote buttons to work, and in the FS coding only use some of the prefixes. I also made the .slides element somewhat clearer to show the Safari issue better.

```
<!DOCTYPE html>

<head>
    <meta charset="utf-8">
    <title>reveal.js - Multiple Presentations</title>
    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js@4.2.0/dist/reveal.css">
    <link id="theme" rel="stylesheet" href="https://cdn.jsdelivr.net/npm/reveal.js@4.2.0/dist/theme/white.css">
    <style>
        html {
            width: 100%;
            height: 100%;
            margin: 0;
            position: relative;
            display: block;
        }

        body {
            background: #ddd;
            width: 100%;
            height: 100%;
            margin: 0;
            position: absolute;
            top: 0;
            left: 0;
            z-index: 1;
            display: block;
        }

        .row {
            display: flex;
            flex-direction: row;
            height: 60vh;
        }

        .col {
            width: 100%;
            height: 100%;
            padding: 2rem;
            display: flex;
            flex-direction: column;
            align-items: center;
        }

        .btn {
            border: 1px solid black;
            padding: .5rem .75rem;
            border-radius: .25rem;
            margin-top: 1rem;
            text-decoration: none;
            color: black;
        }

        .reveal {
            width: 100%;
            height: 100%;
            border: 1px solid black;
        }

        .reveal .slides {
            border: 4px solid #ccc;
        }

        .reveal.focused .slides {
            border-color: #94b5ff;
        }
    </style>
</head>

<body>
    <div class="row">
        <div class="col">
            <p>Before</p>
            <div class="reveal deck1" id="deck1">
                <div class="slides">
                    <section>
                        <h1>Deck 1</h1>
                    </section>
                    <section>Deck 1, Slide 1</section>
                    <section>Deck 1, Slide 2</section>
                    <section>Deck 1, Slide 3</section>
                </div>
            </div><a class="btn" href="#" data-viewdemo="deck1">Demo</a>
            <p>After clicking "Demo", keyboard navigation does NOT work immediately</p>
        </div>
        <div class="col">
            <p>After</p>
            <div class="reveal deck2" id="deck2">
                <div class="slides">
                    <section>
                        <h1>Deck 2</h1>
                    </section>
                    <section>Deck 2, Slide 1</section>
                    <section>Deck 2, Slide 2</section>
                    <section>Deck 2, Slide 3</section>
                </div>
            </div><a class="btn" href="#" data-viewdemo="deck2">Demo</a>
            <p>After clicking "Demo", keyboard navigation works immediately</p>
        </div>
    </div>
    <script src="https://cdn.jsdelivr.net/npm/reveal.js@4.2.0/dist/reveal.min.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/screenfull@5.2.0/dist/screenfull.min.js"></script>
    <script>
        let storedKeyboardCondition = null;

        const fullscreenchange = function(event, deck) {
            if (document.webkitFullscreenElement || document.fullscreenElement) {
                storedKeyboardCondition = deck.getConfig().keyboardCondition;
                deck.configure({
                    keyboardCondition: null
                });
            }
            if (!document.webkitFullscreenElement && !document.fullscreenElement) {
                deck.configure({
                    keyboardCondition: storedKeyboardCondition
                });
            }
        };

        let viewdemos = document.querySelectorAll("[data-viewdemo]");
        viewdemos.forEach(viewdemo => {
            let toBeFSed = viewdemo.getAttribute("data-viewdemo");
            toBeFSed = document.getElementById(toBeFSed);
            viewdemo.addEventListener('click', (e) => {
                if (screenfull.isEnabled) {
                    e.preventDefault();
                    screenfull.request(toBeFSed);
                }
            });
        });


        let deck1 = new Reveal(document.querySelector('.deck1'), {
            embedded: true,
            progress: false,
            keyboardCondition: 'focused'
        });
        deck1.on('slidechanged', () => {
            console.log('Deck 1 slide changed');
        });
        deck1.initialize();

        let deck2 = new Reveal(document.querySelector('.deck2'), {
            embedded: true,
            progress: false,
            keyboardCondition: 'focused'
        });
        deck2.initialize().then(() => {
            deck2.slide(1);
        });
        deck2.on('slidechanged', () => {
            console.log('Deck 2 slide changed');
        });

        deck2.on('fullscreenchange', e => {
            fullscreenchange(e, deck2)
        });
        deck2.on('webkitfullscreenchange', e => {
            fullscreenchange(e, deck2)
        });
    </script>
</body>
```

**Comment by Martinomagnifico** [comment]
In the above presentation, entering fullscreen can be done by either pressing F, or clicking the 'demo' button.

Can any Safari user please confirm that in the _left_ presentation (of the above code), or on _all_ embedded presentations on revealjs.com,  the first slide will not scale to fullscreen initially?

**Comment by hakimel** [comment]
I can confirm, the left example doesn't get scaled up as it should when entering fullscreen in Safari.

**Comment by hakimel** [comment]
Thanks for digging into this, Martino. I've pushed a fix that should resolve all of the issues in b7d65be051072c4f36334403f0cdc19f88ce09c3

Fix for issue 1 & 2:
We now use `fullscreenchange` to trigger a layout AND focus the presentation. This ensures the presentation gains keyboard focus no matter where you enter fullscreen from. This has been written to support multiple embeds, so it will only layout and focus the deck that you take fullscreen even if there are multiple decks on the same page.

Fix for issue 3:
I added new styles that force the presentation to cover the full page when it's in `:fullscreen` mode. This way there's no need to worry about styles from the page affecting the fullscreen mode.

Let me know if you still notice any issues at all. Will update the reveal.js version on revealjs.com soon to fix that as well.

**Comment by Martinomagnifico** [comment]
Thanks Hakim! All Safari users are happy now :-)

**Comment by Martinomagnifico** [comment]
I needed to add "transform: none !important;" to the .reveal-viewport:fullscreen rule, because else the controls are pushed a bit offscreen.

**Comment by hakimel** [comment]
I've added a reset for transform too.

Love that page for your reveal.js plugins! Would be great to link to that somewhere from the revealjs.com docs. Not sure where it fits but maybe a new page dedicated to third party plugins (I'm linking out to the wiki right now). Will give it some thought.
